Source/UserManagement/Domain/StaffUser/StaffUser.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Concepts;
+using doLittle.Domain;
+using doLittle.Runtime.Events;
+using Events.StaffUser;
+namespace Domain.StaffUser
+{
+    public class StaffUser : AggregateRoot
+    {
+        protected StaffUser(EventSourceId id) : base(id) {}
+
+        //TODO: I have no idea, not even sure what AggregateRoot means. Someone should look over this.
+        public void AddStaffUser(AddStaffUser command)
+        {
+            switch (command.Role)
+            {
+                case Role.Admin:
+                    HandleAdmin(command);
+                    break;
+                case Role.DataConsumer:
+                    HandleDataConsumer(command);
+                    break;
+                case Role.DataCoordinator:
+                    HandleDataCoordinator(command);
+                    break;
+                case Role.DataOwner:
+                    HandleDataOwner(command);
+                    break;
+                case Role.DataVerifier:
+                    HandleDataVerifier(command);
+                    break;
+                case Role.SystemCoordinator:
+                    HandleSystemCoordinator(command);
+                    break;
+            }
+        }
+
+        private void HandleAdmin(AddStaffUser command)
+        {
+            Apply(new AdminAdded(
+                Guid.NewGuid(), command.FullName,
```

Comment:
The Id of the Admin to be created should be on the Command.   We don't return any state from the CommandHandler so if we wanted to navigate to the newly created Admin we wouldn't be able to.  If the Guid is generated on the front end as part of the Command, we could then navigate to that Guid given a successful CommandResult.

Where a legacy system creates an auto-generated incremental key (think IDENTITY in Sql Server) for example, I tend to add Guid as a surrogate key and use that for the initial navigation and then the system generated key from then on.